staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go

```diff
@@ -235,6 +235,16 @@ type ValidationRule struct {
 	// If unset, the message is "failed rule: {Rule}".
 	// e.g. "must be a URL with the host matching spec.host"
 	Message string `json:"message,omitempty" protobuf:"bytes,2,opt,name=message"`
+	// MessageExpression declares a CEL expression that evaluates to the validation failure message that is returned when this rule fails.
+	// Since MessageExpression is used as a failure message, it must evaluate to a string.
+	// If both Message and MessageExpression are present on a Rule, then MessageExpression will be used if validation
+	// fails. If MessageExpression results in a runtime error, the runtime error is logged, and the validation failure message is produced
+	// as if the MessageExpression field were unset.
+	// MessageExpression has access to all the same variables as the Rule; the only difference is the return type.
```

Comment:
use serialized field names in godoc (`messageExpression`,`message`, `rule`, etc) so it matches what users actually set via the API

Reply:
Gotcha, I'll change that and mark as resolved when I do.

Reply:
(marking as resolved - think it's worth pointing out that I kept the first mention unchanged from `MessageExpression` since this seems consistent with godoc rules + usage in the rest of the file; if I misunderstood let me know and I can fix that)